### PR TITLE
Orphaned disks removal prowjob

### PR DESCRIPTION
**Pull request by Tomasz-Smelcerz-SAP**
**Description**

Changes proposed in this pull request:

- Introduce a periodic job to clean up orphaned disks every hour.

Note: Default disk cleanup policy is to remove disks that are unused and created more than two hours ago.

**Related issue(s)**
See also #273

### CODE REVIEW

File: prow/jobs/kyma/kyma-integration.yaml [hunk 1]
```diff
@@ -109,10 +109,29 @@ periodics:
...
-  cron: "0 7 * * 1-5" # “At 07:00 on every day-of-week from Monday through Friday.”
+  cron: "0 7 * * 1-5" # "At 07:00 on every day-of-week from Monday through Friday."
   labels:
     preset-sa-vm-kyma-integration: "true"
     preset-gc-project-env: "true"
   spec:
     containers:
     - image: eu.gcr.io/kyma-project/prow/cleaner:0.0.1 # see test-infra/prow/images/cleaner
+- name: orphanded-disks-cleaner
+  cron: "0 * * * *" # "Every hour"
```
Comment (Demonsthere) [inline prow/jobs/kyma/kyma-integration.yaml]: Are you sure every hour is necessary? @aszecowka @michal-hudy what do you think, how often should it run?

File: development/resources-cleanup.sh [hunk 2]
```diff
@@ -0,0 +1,42 @@
+#!/usr/bin/env bash
```
Comment (Demonsthere) [inline development/resources-cleanup.sh]: I would change the name of the script to something more straightforward like `disk-cleanup.sh`. We need to clean multiple resources, so we will probably have a few of scripts like this.

File: development/resources-cleanup.sh [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+echo "Removing GCP disk resources allocated by failed/terminated integration jobs..."
+
+if [ ! -d "${DEVELOPMENT_DIR}/tools/vendor" ]; then
+    echo "Vendoring 'tools'"
+    pushd "${DEVELOPMENT_DIR}/tools"
+    dep ensure -v -vendor-only
+    popd
+fi
+
+go run "${DEVELOPMENT_DIR}"/tools/cmd/diskscollector/main.go  --project="${PROJECT}" --dryRun=true
```
Comment (sjanota) [inline development/resources-cleanup.sh]: I guess it should be `--dryRun=false` ?

File: development/resources-cleanup.sh (hunk 2 above)
Comment (Tomasz-Smelcerz-SAP) [reply development/resources-cleanup.sh]: Renamed to disks-cleanup.sh

File: prow/jobs/kyma/kyma-integration.yaml (hunk 1 above)
Comment (Tomasz-Smelcerz-SAP) [reply prow/jobs/kyma/kyma-integration.yaml]: Do you think it's too often? What's the suggested value then?

Comment (Demonsthere) [reply prow/jobs/kyma/kyma-integration.yaml]: I would start with once a day and go from there increasing the frequency as required

Comment (Tomasz-Smelcerz-SAP) [reply prow/jobs/kyma/kyma-integration.yaml]: I personally prefer to remove it ASAP, after all, we pay for these resources. It means they should be removed once they're not needed. One hour is not particularly fast but is fast enough to cut the costs down.

File: prow/jobs/kyma/kyma-integration.yaml [hunk 4]
```diff
@@ -109,10 +109,29 @@ periodics:
...
+      base_ref: master
+      path_alias: github.com/kyma-project/test-infra
+  spec:
+    containers:
+    - image: eu.gcr.io/kyma-project/prow/buildpack-golang:0.0.1
+      command:
+        - "bash"
+      args:
+        - "-c"
+        - "development/disks-cleanup.sh ${CLOUDSDK_CORE_PROJECT} --dryRun=false"
```
Comment (jakkab) [inline prow/jobs/kyma/kyma-integration.yaml]: disks-cleanup.sh script seems to accept but one argument, so the "--dryRun=true" flag is not going to affect the way it works, I guess.

Comment (Tomasz-Smelcerz-SAP) [reply prow/jobs/kyma/kyma-integration.yaml]: You're right. It's redundant.

**Pull request merged**